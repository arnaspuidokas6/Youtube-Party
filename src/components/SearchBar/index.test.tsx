import React from 'react';
import { render, screen } from '@testing-library/react';
import { SearchBar } from '.';
import userEvent from '@testing-library/user-event';

describe('SearchBar', () => {
    it('should render', () => {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        const { container } = render(<SearchBar />);

        expect(screen.getByTestId('search-input')).toBeVisible();
        expect(container).toMatchSnapshot();
    });

    it('should validate too long input', () => {
        render(<SearchBar />);
        userEvent.type(screen.getByTestId('search-input'), '123445557889900001111222233344');
        expect(screen.getByText('Invalid search!')).toBeVisible();
    });
});
