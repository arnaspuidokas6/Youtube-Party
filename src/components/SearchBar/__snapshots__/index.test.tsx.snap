// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<div>
  <div
    class="w-screen flex flex-row p-8 items-center justify-between bg-white shadow-xs border solid"
  >
    <div
      class="text-lg text-gray-700 hidden md:flex"
    >
      Fake Youtube App
    </div>
    <span
      class="w-screen md:w-1/3 h-10 bg-gray-200 cursor-pointer rounded-full"
    >
      <div
        class="relative"
      >
        <input
          class="sm:text-base relative w-full rounded-full border focus:outline-none py-2 pr-2 pl-12 "
          data-testid="search-input"
          name="search"
          placeholder="Search"
          type="search"
        />
      </div>
    </span>
  </div>
</div>
`;
